File: CodeWars/Algorithms/WhichAreIn/WhichAreIn.rs

fn in_array(arr1: Vec<&str>, arr2: Vec<&str>) -> Vec<String> {
    let mut temp: Vec<String> = arr1.iter()
        .filter(|&s| arr2.iter().any(|&t| t.contains(s)))
        .map(|s| s.to_string()).collect();
    temp.sort();

    temp
}

fn dotest(arr1: Vec<&str>, arr2: Vec<&str>, exp: Vec<&str>) {
    assert_eq!(exp, in_array(arr1, arr2));
}

fn main() {
    dotest(vec!["arp", "live", "strong"],
           vec!["lively", "alive", "harp", "sharp", "armstrong"],
           vec!["arp", "live", "strong"]);

    dotest(vec!["cod", "code", "wars", "ewar", "pillow", "bed", "phht"],
           vec!["lively", "alive", "harp", "sharp", "armstrong", "codewars", "cod", "code"],
           vec!["cod", "code", "ewar", "wars"]);
}

File: CodeWars/Algorithms/WhichAreIn/WhichAreIn.py
import unittest

def in_array(arr1, arr2):
    return []


class TestExercise(unittest.TestCase):
    def test_in_array(self):
        a1 = ["live", "arp", "strong"] 
        a2 = ["lively", "alive", "harp", "sharp", "armstrong"]
        r = ['arp', 'live', 'strong']
        self.assertEqual(in_array(a1, a2), r)

if __name__ == '__main__':
    unittest.main()
